arch/x86/core/pcie.c

```diff
@@ -262,8 +262,8 @@ uint8_t arch_pcie_msi_vectors_allocate(unsigned int priority,
 				return 0;
 			}
 
-			for (i = irte; i < (irte + n_vector); i++) {
-				vectors[i].arch.irte = i;
+			for (i = 0; i < n_vector; i++) {
+				vectors[i].arch.irte = irte + i;
```

Comment:
which is .... the same? 
Useless change.

Reply:
this change is to address a bug in existing code which might cause hard to debug issue due to array boundary overflow(lead to memory corruption). With existing code, driver cannot allocate interrupt beyond what driver requesting vector size(n_vector) which will crash the code (vector[] buffer index overflow).

Reply:
oh yes true, that was fixed here https://github.com/zephyrproject-rtos/zephyr/pull/36731/commits/9acfbfcd284f213fc72db3a0a0eb9fde496148c4